src/model/scripts/lighting_training/train.py

```diff
@@ -59,9 +59,9 @@
     )
     parser.add_argument(
         "--checkpoint",
-        type=bool,
-        default=False,
-        help="gradient checkpoint only available for GPTJ",
+        type=int,
+        default=0,
+        help="0 | 1 (model) | 2 (self-attentions only)",
```

Comment:
I think we should have 2 parameters. 1 is for `use_checkpointing` and 1 is for `checkpointing_position`. Is it better than 0, 1, 2?

Reply:
done